platform_plugin_aspects/views.py

```diff
@@ -122,3 +140,99 @@ def get(self, request, *args, **kwargs):
             raise APIException() from exc
 
         return Response({"guestToken": guest_token})
+
+
+class SupersetInContextDashboardView(GenericAPIView):
+    """
+    Endpoint for in-context analytics embedded Superset dashboard parameters.
+    """
+
+    authentication_classes = (SessionAuthentication,)
+    permission_classes = (
+        permissions.IsAuthenticated,
+        IsStaffOrReadOnly | IsCourseStaffInstructor,
+    )
+
+    lookup_field = "usage_id"
+
+    def get_object(self):
+        """
+        Return a usage key or course key for the requested usage_key.
+        """
+        usage_id = self.kwargs.get(self.lookup_field, "")
+        try:
+            usage_key = UsageKey.from_string(usage_id)
+        except InvalidKeyError as exc:
+            try:
+                course_key = usage_key = CourseKey.from_string(usage_id)
+            except InvalidKeyError:
+                raise NotFound(
+                    _("Invalid usage id: '{usage_id}'").format(usage_id=usage_id)
+                ) from exc
+        else:
+            course_key = usage_key.course_key
+
+        course = _get_course(course_key)
+        self.check_object_permissions(self.request, course)
+
+        return usage_key
+
+    def get(self, request, *args, **kwargs):
+        """
+        Return Superset context for embedding the dashboard for the requested block.
+        """
+        usage_key = self.get_object()
+        if isinstance(usage_key, CourseKey):
+            block_type = "course"
+            course_key = usage_key
+        else:
+            block_type = usage_key.block_type
+            course_key = usage_key.course_key
+
+        course_key_col = settings.IN_CONTEXT_DASHBOARD_COURSE_KEY_COLUMN
+        block_id_col = settings.IN_CONTEXT_DASHBOARD_BLOCK_ID_COLUMN
+
+        dashboards = settings.ASPECTS_IN_CONTEXT_DASHBOARDS
+        dashboard = dashboards.get(block_type)
+        if dashboard is None:
+            raise NotFound(
+                _("No dashboard for block type: '{block_type}'").format(
+                    block_type=block_type
+                )
+            )
+
+        block_filter = {}
+        if not isinstance(usage_key, CourseKey):
+            for filter_id in dashboard["block_filter_ids"]:
+                block_filter[filter_id] = build_filter(
+                    filter_id, block_id_col, "IN", [usage_key.block_id]
+                )
+
+        course_runs = {}
+        # TODO: obtain course runs.
+        for course_run in [course_key.run]:
+            course_filter = block_filter.copy()
+            for filter_id in dashboard["course_filter_ids"]:
+                course_filter[filter_id] = build_filter(
+                    filter_id, course_key_col, "IN", [str(course_key)]
+                )
+            course_runs[course_run] = {
+                "native_filters": prison.dumps(course_filter),
```

Comment:
This usage of `prison` probably deserves a comment since I think it's the only place in this repo where it's used.